DataProviderService/StarSystemSqLiteRepository.cs

```diff
@@ -197,19 +193,14 @@ public List<StarSystem> GetOrFetchStarSystems(string[] names, bool fetchIfMissin
 
         public StarSystem GetStarSystem(string name, bool refreshIfOutdated = true)
         {
-            if (String.IsNullOrEmpty(name)) { return null; }
-            return GetStarSystems(new[] { name }, refreshIfOutdated)?.FirstOrDefault();
+            return string.IsNullOrEmpty(name) ? null : GetStarSystems(new[] { name }, refreshIfOutdated)?.FirstOrDefault();
         }
 
         public List<StarSystem> GetStarSystems(string[] names, bool refreshIfOutdated = true)
         {
-            if (!File.Exists(DbFile))
-            {
-                return null;
-            }
-            if (!names.Any()) { return null; }
-
             List<StarSystem> results = new List<StarSystem>();
+            if (!File.Exists(DbFile) || !names.Any()) { return results; }
```

Comment:
This one is a bit less clear-cut but after considering it for a bit I think I still prefer the two separate if statements because they are guarding against two unrelated circumstances. But yes hoist the definitionf of `results` and return that rather than `null`.